face_autoencoder/image_processing.py

```diff
@@ -21,3 +23,37 @@ def __call__(self, image: np.ndarray) -> torch.Tensor:
 
     def preprocess_image(self, image: np.ndarray) -> torch.Tensor:
         return self.image_preprocessing_pipeline(Image.fromarray(image))
+
+
+class PreprocessingPipelineWithEdges:
+    def __init__(self, nn_input_image_resolution: int, weights: int) -> None:
+        self.nn_input_image_resolution = nn_input_image_resolution
+        self.weights = weights
+        self.image_preprocessing_pipeline = transforms.Compose(
+            [
+                transforms.ToTensor(),
+                transforms.Resize(
+                    (nn_input_image_resolution, nn_input_image_resolution)
+                ),
+            ]
+        )
+
+    def __call__(
+        self, image: np.ndarray, edge: np.ndarray
+    ) -> t.Tuple[torch.Tensor, torch.Tensor]:
+        return self.preprocess_image(image, edge)
+
+    def preprocess_image(
+        self, image: np.ndarray, edge: np.ndarray
+    ) -> t.Tuple[torch.Tensor, torch.Tensor]:
+        edge = self.image_preprocessing_pipeline(Image.fromarray(edge))
+        # edge = edge / 255 * self.weights
+        for i in range(0, edge.shape[0]):
+            for j in range(0, edge.shape[1]):
+                for k in range(0, edge.shape[2]):
+                    if edge[i, j, k] == 0:
+                        edge[i, j, k] = 1
+                    else:
+                        edge[i, j, k] = self.weights
```

Comment:
```suggestion
     edge[edge == 255] = self.weights
     edge[edge == 0] = 1
```

Reply:
Do tego fragmentu musze wrócić, bo przez "edge = self.image_preprocessing_pipeline(Image.fromarray(edge))" wynikowy tensor nie składa się z 0 i 255, a z 0 i np. 0.25, 0.5 i 0.75. Teraz jest: edge[edge != 0] = self.weights
        edge[edge == 0] = 1, ale nie wiem czy edge != 0 jest równoznaczne edge == 255 przed preprocessingiem

Reply:
nie mozemy przepuscic krawedzi przez ten pipeline bo tam jest normalizacja. Krawedzie trzeba tylko przeskalowac i zaminic na tensor i to wszystko